enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,281 @@
+  pattern.
+  - Users still have the option of creating a new `AlertingRule` resource based
+    on an existing platform rule, making their modifications there, and
+    silencing the original alert.
+
+## Proposal
+
+The cluster-monitoring-operator will be extended with two new resources types:
+`AlertingRule` and `AlertRelabelConfig`, both in the `monitoring.openshift.io`
+group.
+
+The new `AlertingRule` type will allow additional user-defined alerting rules,
+but without the ability to create recording rules provided by the existing
+`PrometheusRule` resource type. Example:
+
+```yaml
+apiVersion: v1alpha1
+kind: alertingrule.monitoring.openshift.io
+metadata:
+  name: custom-alerts
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: CustomGroup
+    rules:
+    - alert: CustomAlert
+      expr: "vector(1)"
+      for: 15m
+      labels:
+        custom: "true"
+```
+
+A new controller in the cluster-monitoring-operator will generate
+`PrometheusRule` resources based on the `AlertingRule` instances in the
+`openshift-monitoring` namespace.  These will be consumed by the platform
+prometheus-operator instance like any other `PrometheusRule`.  All user-defined
+alerts will be labeled identifying them as such -- likely using the existing
+`openshift_io_alert_source` label.
+
+The new `AlertRelabelConfig` type will allow relabeling operations on alerts.
+All `AlertRelabelConfig` instances will be collected in order to build a single
+secret with additional alert relabel configs used by the prometheus-operator.
+Example of changing an alert severity from `none` to `critical`:
+
+```yaml
+apiVersion: v1alpha1
+kind: alertrelabelconfig.monitoring.openshift.io
+metadata:
+  name: severity-change
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: SeverityChanges
+    configs:
+    - source_labels: [alertname, severity]
+      regex: "Watchdog;none"
+      target_label: severity
+      replacement: critical
+      action: replace
+```
+
+A new controller in the cluster-monitoring-operator will combine all
+`AlertRelabelConfig` instances in the `openshift-monitoring` namespace into a
+single secret, concatenating each group in lexigraphical order by name.  This
+secret will be referenced in the `additionalAlertRelabelConfigs` parameter for
+the platform Prometheus instance.
+
+The `AlertRelabelConfig` controller may optionally disable certain actions like
+`drop` to prevent users disabling critical platform alerts.  Silences already
+exist as a method of ignoring unwanted alerts.
+
+### User Stories
+
+- As an OpenShift user, I want to change the severity of an existing alert,
+  choosing between, e.g. from `Warning` to `Critical`.
+- As an OpenShift user, I want to add labels to alerts to aide in routing and
+  triage.
+- As an OpenShift user, I want a supported way to add custom alerts using
+  metrics from components in the `openshift-*` namespaces.
+
```

Comment:
Maybe add something about alerts that should still be forwarded to telemetry

> As an OpenShift developer, I want that the alerting rules shipped by my component are still evaluated and forwarded to telemetry even when overridden by the users.

Reply:
Done.